GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -428,45 +428,29 @@ public virtual void GetFileTypeAndMode(string filePath, out FileType fileType, o
             }
         }
 
-        public virtual PathSparseState GetPathSparseState(string virtualPath)
+        public virtual PathSparseState GetFolderPathSparseState(string virtualPath)
         {
-            this.GetChildNameAndParentKey(virtualPath, out string fileName, out string parentKey);
-            FolderEntryData data = this.GetProjectedFolderEntryData(
-                blobSizesConnection: null,
-                childName: fileName,
-                parentKey: parentKey);
-
-            if (data == null)
-            {
-                return PathSparseState.NotFound;
-            }
-            else if (data.IsIncluded)
-            {
-                return PathSparseState.Included;
-            }
-            else
+            if (this.TryGetFolderDataFromTreeUsingPath(virtualPath, out FolderData folderData))

```

Comment:
Similar question as in `TryAddSparseFolder`, why was this changed from `GetProjectedFolderEntryData` to `TryGetFolderDataFromTreeUsingPath`?

Reply:
`GetProjectedFolderEntryData` will not return entries that are not included and we we need to be able to return `Excluded` entries.